internal/provider/client.go

```diff
@@ -42,14 +45,38 @@ func NewClient(host string, username *string, password *string, insecureSkipVeri
 	}
 	client.httpClient = &http.Client{Transport: tr, Timeout: time.Duration(timeout) * time.Second}
 
-	return &client, nil
+	body, diags := client.Get("/api/")
```

Comment:
I think what we really need to do is support accessing controller directly (AAP 2.4) and accessing controller through the API gateway (AAP 2.5+). What we have here works for controller directly, but not for gateway. We'll probably need to assume gateway, first, then if that fails, fallback to controller directly. Here's what I get when I access `/api/` on gateway:
```json
{
  "description": "AAP gateway REST API",
  "apis": {
    "gateway": "/api/gateway/",
    "controller": "/api/controller/",
    "eda": "/api/eda/",
    "galaxy": "/api/galaxy/"
  }
}
```
A further fetch of `/api/controller/` gives:
```json
{
  "description": "AWX REST API",
  "current_version": "/api/controller/v2/",
  "available_versions": {
    "v2": "/api/controller/v2/"
  },
  "custom_logo": "",
  "custom_login_info": "",
  "login_redirect_override": ""
}
```

Reply:
updated